test/e2e/framework/clusterpropagationpolicy.go

```diff
@@ -61,23 +61,29 @@ func PatchClusterPropagationPolicy(client karmada.Interface, name string, patch
 // UpdateClusterPropagationPolicyWithSpec update PropagationSpec with karmada client.
 func UpdateClusterPropagationPolicyWithSpec(client karmada.Interface, name string, policySpec policyv1alpha1.PropagationSpec) {
 	ginkgo.By(fmt.Sprintf("Updating ClusterPropagationPolicy(%s) spec", name), func() {
-		newPolicy, err := client.PolicyV1alpha1().ClusterPropagationPolicies().Get(context.TODO(), name, metav1.GetOptions{})
-		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-
-		newPolicy.Spec = policySpec
-		_, err = client.PolicyV1alpha1().ClusterPropagationPolicies().Update(context.TODO(), newPolicy, metav1.UpdateOptions{})
-		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Eventually(func() error {
+			newPolicy, err := client.PolicyV1alpha1().ClusterPropagationPolicies().Get(context.TODO(), name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			newPolicy.Spec = policySpec
+			_, err = client.PolicyV1alpha1().ClusterPropagationPolicies().Update(context.TODO(), newPolicy, metav1.UpdateOptions{})
+			return err
+		}, PollTimeout, PollInterval).ShouldNot(gomega.HaveOccurred())
```

Comment:
This pattern would slow down the test from reporting real errors until the timeout is reached. Concerning.
For instance, when you are trying to update a non-existent resource, it is supposed to fail quickly, because that usually means something is wrong with the test steps.